### ISSUE TITLE
Error in annotateSomaticQs(variants$variants, genome = genome, resourceDirectory = resourceDirectory,  :    NA x or somaticP or variant in annotateSomaticQs.

### ISSUE THREAD

**Issue by jlac** [ChristofferFlensburg/superFreq#57]
Hi,

I am encountering the following error running a split run of SuperFreq:

Error in annotateSomaticQs(variants$variants, genome = genome, resourceDirectory = resourceDirectory,  : 
  NA x or somaticP or variant in annotateSomaticQs.
Calls: superFreq -> superFreq -> analyse -> annotateSomaticQs
Execution halted

I can see that it appears to be related to some unexpected value for a variant or class, but i can't seem to figure out what's causing it. For the variants supplied for this analysis, the normal sample is a VCF generates by GATK germline calling, and the VCFs for the tumor samples (there are 5 tumor samples) were merged from mutect, mutect2, and strelka. This approach has worked for 40 other patients in the cohort, it is just this one patient that seems to be failing. I am happy to supply any files that might be needed for diagnosing this issue. Thanks in advance for your help!

Justin

**Comment by ChristofferFlensburg** [comment]
Hi!

You are right that it seems to be an upstream error resulting in NA values that shouldn't be NA. If it worked for the others, it seems like either a corner case bug, or there are some issues with the input data for that sample.

I can't tell from error message alone, but if you send me the log file (Rdirectory/patientName/runtimeTracking.log) I might find some clues. Maybe also send me a log file from one of the individuals that did go through, so I can compare.

Also maybe you can have a quick look at the input files and check if there are any glaring issues, such as empty files or whatever. Your setup with the germline variants to the matched normal and somatic variants to the others should be ok.

FYI, that file includes the meta data you supplied, but no identifying germline variants. If you don't feel comfortable sharing your metadata publicly (I think it's safe, don't think anyone else will download it, but up to you and the data generators), then you can also email it to me at flensburg.c at wehi.edu.au.

**Comment by jlac** [comment]
I looked over each input VCF and BAM file and I don't see any glaring issues. No files are empty, and all files for the successful samples were generated in exactly the same way with the same workflow as for the the sample that failed.

I am attaching the log files for the failed sample and for a successful sample here. I should also have said in my earlier message that when generating the somatic VCFs for the tumor samples (for all patients) that I retained all variants in the VCFs, including all that did not PASS the internal filters for each individual caller. This seemed appropriate given the instructions in the README, but let me know if additional upstream filtering might be appropriate.

Thank you!
[completed_runtimeTracking.log](https://github.com/ChristofferFlensburg/superFreq/files/4963644/completed_runtimeTracking.log)
[failed_runtimeTracking.log](https://github.com/ChristofferFlensburg/superFreq/files/4963645/failed_runtimeTracking.log)

**Comment by ChristofferFlensburg** [comment]
Thanks for sending those through!

It looks fine up to the point where the somaticP score is calculated, no signs of issues with with input. Then:
```
Marking somatic mutations in Pt29_4049_A_1.normal..
No matched normal, or normal sample: selecting somatic variants based on population frequencies. Selecting dbSNPs and ExAC below 0.1% population frequency as somatic candidates. These will include rare germline variants, which is desired for normals, but not for cancer samples without matched normals.
Salvaging  0  sites that have high frequency in dbSNP or ExAC, but also high frequency in COSMIC.
got roughly 1074 somatic variants.
Marking somatic mutations in Pt29_4049_A_1..Salvaging  0  sites that have high frequency in dbSNP or ExAC, but also high frequency in COSMIC.
Correcting somatics using Pt29_4049_A_1.normal as matched normal.
got roughly NA somatic variants.
Marking somatic mutations in Pt29_4049_B_1..Salvaging  0  sites that have high frequency in dbSNP or ExAC, but also high frequency in COSMIC.
Correcting somatics using Pt29_4049_A_1.normal as matched normal.
got roughly 71 somatic variants.
Marking somatic mutations in Pt29_4049_C_1..Salvaging  0  sites that have high frequency in dbSNP or ExAC, but also high frequency in COSMIC.
Correcting somatics using Pt29_4049_A_1.normal as matched normal.
got roughly NA somatic variants.
Marking somatic mutations in Pt29_4049_D_1..Salvaging  0  sites that have high frequency in dbSNP or ExAC, but also high frequency in COSMIC.
Correcting somatics using Pt29_4049_A_1.normal as matched normal.
got roughly 65 somatic variants.
Marking somatic mutations in Pt29_4049_E_1..Salvaging  0  sites that have high frequency in dbSNP or ExAC, but also high frequency in COSMIC.
Correcting somatics using Pt29_4049_A_1.normal as matched normal.
got roughly 146 somatic variants.
```
two of the samples got at least one NA somaticP score (the "roughly" is the sum of all somaticP for that sample). I'd need to look closer at those variants to figure out what's going on. As we don't know at this point whether the NA scores come from somatic or germline variants, I think you should reply to the email above to avoid posting patient germline information publicly.

If you are familiar with R, the easiest would be if you extract them yourself and just copy paste the output to me in an email:
``` R
#load in variants from last savepoint before crash
load('path/to/superfreq/Pt29/R/Pt29/allVariantsPreVEP.Rdata') #fill in path here, depending where you run R from.

#extract data frame with all variants of one of the problem samples
q = allVariants$variants$variants$Pt29_4049_C_1

#count number of somatic scores that are NA (please include in email)
cat('Number of NA somaticP: ', sum(is.na(q$somaticP)), '\n')

#print some NA variants (include in email)
print(q[is.na(q$somaticP),][1:5,])
```

Is that ok?

Thanks.

**Comment by ChristofferFlensburg** [comment]
Thanks for email.

There was only one NA somaticP in the problem sample. The variant had 0 reads covering the site, which I first thought was the cause of the error, but looking at the code, superFreq should handle that (and indeed does when I tested it). Either way, I added a check for NA somaticP, and it'll set them to 0 with a warning with some more diagnostics in the log file. Just pushed live.

So update to version 1.4.1, and rerun. Hopefully that should go through. Confirm that there is a warning about NA somaticP, and send me the logfile again so that I can have a look at the diagnostics that hopefully should be there. If I can figure out the cause, then I might fix it, or if not, I'll just leave it as it is with the warning.

Let me know how it goes!
